Fix linting

chombium opened the pull request:
I've run golangci-lint and fixed all of the findings.

Remarks to these change:
- there are many functions which return error and are called in defer functions or run as goroutines. I've added a a comment to  ignored these findings
- there are many functions which are not used which I've commented out as I'm not sure if they should be kept or not
- There is a [RefreshTokenKeys](https://github.com/cloudfoundry/log-cache-release/blob/main/src/internal/auth/uaa_client.go#L88-L173) function in `uaa_client.go` which is called once as a [goroutine](https://github.com/cloudfoundry/log-cache-release/blob/main/src/internal/auth/uaa_client.go#L220) and once [directly](https://github.com/cloudfoundry/log-cache-release/blob/main/src/internal/auth/uaa_client.go#L255). Adding error check to the "normal" call makes few tests break as there are some timing issues. IMO it would we good if we refactor this in order to avoid problems. Do you thing this worth taking a look?

Fixes #97 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing performed?

- [x] Unit tests
- [ ] Integration tests
- [ ] Acceptance tests

## Checklist:

- [x] This PR is being made against the `main` branch, or relevant version branch
- [ ] I have made corresponding changes to the documentation
- [ ] I have added testing for my changes

ctlong commented on src/internal/auth/cf_auth_middleware.go:
```diff
@@ -117,11 +117,14 @@ func (m CFAuthMiddlewareProvider) Middleware(h http.Handler) http.Handler {
 				Error:     err.Error(),
 			})
+			if err != nil {
+				log.Printf("bad data: %v", err)
```
Can you please add a `\n` to the end of this string?

ctlong commented on src/internal/auth/cf_auth_middleware_test.go:
```diff
@@ -397,9 +397,9 @@ var _ = Describe("CfAuthMiddleware", func() {
-	user          string
-	err           error
+	// client        string
+	// user          string
```
If they're unused, please go ahead and delete these variables :)

ctlong commented on src/internal/auth/uaa_client_test.go:
```diff
@@ -107,21 +105,26 @@ var _ = Describe("UAAClient", func() {
-			tc.uaaClient.Read(withBearer(token))
+			_, err := tc.uaaClient.Read(withBearer(token))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(err).ToNot(HaveOccurred())
```
There's a duplicate line here.

ctlong commented on src/internal/auth/uaa_client_test.go:
```diff
@@ -450,15 +461,15 @@ func generateLegitTokenKey(keyId string) mockTokenKey {
+// 		publicKey:  privateKey,
+// 		keyId:      keyId,
+// 	}
+// }
```
If this function is no longer used, then please delete it :)

ctlong commented on src/internal/auth/uaa_client_test.go:
```diff
@@ -691,30 +702,30 @@ func (s *spyHTTPClient) Do(r *http.Request) (*http.Response, error) {
+// 	b := make([]byte, 8)
+// 	binary.LittleEndian.PutUint64(b, uint64(privateKey.PublicKey.E))
+// 	return base64.StdEncoding.EncodeToString(b[0:3])
+// }
```
Please delete these unused functions.

ctlong commented on src/internal/auth/uaa_client_test.go:
```diff
@@ -736,9 +747,9 @@ func keyPEMToString(privateKey *rsa.PrivateKey) (string, string) {
-}
+// func publicKeyModulusToString(privateKey *rsa.PrivateKey) string {
+// 	return base64.StdEncoding.EncodeToString(privateKey.PublicKey.N.Bytes())
+// }
```
Please delete this unused function.

ctlong commented on src/internal/cache/store/store_benchmark_test.go:
```diff
@@ -151,14 +150,14 @@ func BenchmarkMetaWhileReading(b *testing.B) {
+// 	default:
+// 		return false
+// 	}
+// }
```
If unused, please delete.

ctlong commented on src/internal/gateway/gateway.go:
```diff
@@ -179,7 +180,10 @@ func (g *Gateway) listenAndServe() {
-	w.Write([]byte(fmt.Sprintf(`{"version":"%s","vm_uptime":"%d"}`+"\n", g.logCacheVersion, g.uptimeFn())))
+	_, err := w.Write([]byte(fmt.Sprintf(`{"version":"%s","vm_uptime":"%d"}`+"\n", g.logCacheVersion, g.uptimeFn())))
+	if err != nil {
+		g.log.Fatalf("Cannot send result for the info endpoint")
```
Why should this be a fatal log when it wasn't before? Perhaps this should just be an error log.

ctlong commented on src/internal/promql/promql.go:
```diff
@@ -26,7 +26,7 @@ type PromQL struct {
 	instantQueryTimer metrics.Gauge
 	rangeQueryTimer   metrics.Gauge
-	failures          int
+	// failures          int
```
Please delete if unused.

ctlong commented on src/internal/promql/promql_test.go:
```diff
@@ -1370,10 +1370,10 @@ func (s *spyDataReader) ReadEnvelopeTypes() [][]logcache_v1.EnvelopeType {
-}
+// 	s.readResults = es
+// 	s.readErrs = errs
+// }
```
Please delete, if unused.

ctlong reviewed (changes requested):
Thanks for making this PR @chombium! I've requested some changes, please have a look when you get a chance.

chombium commented:
Hi @ctlong,

Thanks for your review. I've fixed all of the issues and re-run the unit tests and linter. Everything looks good on my machine :)

I've check the failing unit tests and compared the logs from he GH Workflow action and my local run and apart from my local run `Starting 16 workers...` and in the action `Starting 4 workers...` there are no other differences.

There is a small caveat in the way the uaa_client is refreshing uaa tokens. The function is  called as a goroutine [here](https://github.com/cloudfoundry/log-cache-release/blob/main/src/internal/auth/uaa_client.go#L220) in uaa_client.Read function and it is called directly many times. I think that there might be some timing issue after I added the error checking. Do you think that it's worth checking the problem or I can simply remove the error checks and add nolint:errcheck directives wherever `uaaClient.RefreshTokenKeys()` is called in a single test where `uaaClient.Read()` is called as well?

ctlong commented:
@chombium thanks for making the changes. Tests are passing in GH actions 🎉 

I think it's okay to remove the error checks and add nolint:errcheck directives for that method of the uaaClient if it's causing problems.

Unfortunately, it appears that there's still one more error in the golangci-lint run by GH actions. Can you please check that out?

chombium commented:
HI @ctlong , I've made the needed changes, now the linting should also be green. For the last finding in the linting checks I've set the MinVersion parameter for TLS, but as I saw that many people are complaining that have this finding as false positive, I've added a `//nolint:gosec` as well, so that it won't be reported anymore.

I've also checked the findings from `go vet`, but I couldn't fix it. I've googled a bit  and found the same function signature for the Seek function in few other projects including Prometheus it self. As there is no way to ignore vet findings, I guess we'll have to live  with it :-/

chombium commented:
@ctlong Coild you please re-trigger the checks? The lint action timmed out loading packages and failed
